### ISSUE TITLE
Azure CLI tasks are far too verbose by default distracting from the actual task results

### ISSUE THREAD

**Issue by ransagy** [microsoft/azure-pipelines-tasks#15845]
*Type*: Feature
**Enter Task Name**: AzureCLI

## Environment
- Server - Azure Pipelines
- Agent - Private, both Windows-latest and Ubuntu-latest
    
    - If using Hosted agent, provide agent queue name:

    - If using private agent, provide the OS of the machine running the agent and the agent version: 

## Issue Description

By default, Azure CLI tasks produce a lot of verbose text including the subscription full details and other information that has nothing to do with what was actually written in the script for the task.

There does not seem to be a way to suppress that, so its very hard to follow what the task's actual output was.
Please consider moving it to verbose/debug text or allowing some flag to "quiet" the output.

**Comment by fowl2** [comment]
It would be great if the log viewer would allow filtering by log level. Probably be a good idea if it would even filter out "info" or "verbose" by default too.

**Comment by TheHokieCoder** [comment]
I agree with @ransagy. By default the task is way too chatty. Microsoft is normally in the habit of making things default to "only what you need to know", with options to increase verbosity. For the AzureCLI task I would recommend that the following steps in the task be suppressed under normal conditions:
- `az --version`
- `az cloud set -n AzureCloud`
- `az login --service-principal -u *** "--password=***" --tenant ********-****-****-****-************ --allow-no-subscriptions`
- `az account set --subscription ********-****-****-****-************`
- `az account clear`
This would leave only the output of the command(s) specified in the task inputs to be displayed, by default. Then, add an input for the task that would allow for showing the output of the other steps that are built in to the task.

**Comment by piotr-gbyliczek** [comment]
I've come across this issue too and attempted to change the code to facilitate hiding extra output: https://github.com/microsoft/azure-pipelines-tasks/pull/16701

**Comment by ransagy** [comment]
Well, it still is a very annoying issue.

**Comment by DanielHRio** [comment]
Anyone can review and approve the PR? Thanks.

**Comment by piotr-gbyliczek** [comment]
If anybody can help, my PR #16701 seems to be stuck waiting endlessly for some CI tasks that I have no idea how to trigger, or if it even should be my action to trigger them. Any advice welcome.

**Comment by vuylstekebram** [comment]
Any updates on this issue? We could really use this in our IaC environment.

**Comment by piotr-gbyliczek** [comment]
@vuylstekebram I have PR open,  #16701 , but can't seem to be able to complete it. There are build tasks marked as required, but I have no idea how to initiate/complete these tasks.

**Comment by ransagy** [comment]
@v-ibshaik @bishal-pdMSFT Can anyone review or at least trigger the CI for that PR? Or assign to someone who can?

**Comment by martyzee** [comment]
Commenting for updates - this would be a great fix

**Comment by TiTi** [comment]
+1

**Comment by BMuuN** [comment]
+1

**Comment by ransagy** [comment]
Mirroring a comment left on the PR mentioned - https://github.com/microsoft/azure-pipelines-tasks/pull/16701#issuecomment-2042407084 - you can control SOME of the output with env vars on the task and some clever use of the flags on each actual script command.
As said, more of a workaround and not a full one, but something.

Perhaps a start would be to remove or mute the `az --version` call, or at least make it adhere to the flag of the diagnostic/debug mode being enabled in the pipeline.

**Comment by TiTi** [comment]
Great ! Thank you @ransagy

**Comment by TheHokieCoder** [comment]
Any update from someone from Microsoft about Piotr's solution/PR?

**Comment by piotr-gbyliczek** [comment]
@TheHokieCoder My solution has been merged nearly a year ago.
